### ISSUE TITLE
[FEATURE] French translations

### ISSUE THREAD

**Issue by JulianMaurin** [tiangolo/fastapi#1972]
# French translations tracking

Bonjour, this is the issue to track and coordinate the French translations. 🇫🇷 
Feel free to add a comment here to indicate the page you plan to translate 📚

Thanks ❤️

**Comment by JulianMaurin** [comment]
I'm starting with the `docs/index.md`, a PR will follow :v:

**Comment by Jefidev** [comment]
Hello, I would like to help with this but it is the first time I contribute to an open-source project. Could you summarize me the workflow to correctly push the update ?

**Comment by Kludex** [comment]
@Jefidev 
- Post here the page you'll be translating.
- Work on it.
- Create the PR with your work.
- Developers will review.
- Fix issues/get approvals.
- Wait for enough approvals (usually 2-3 are enough based on what I've seen).
- @tiangolo will merge if everything is fine.

**Comment by Jefidev** [comment]
Thanks, So I am translating the features.md page.

**Comment by JulianMaurin** [comment]
Hey @Jefidev , to contribute you can also review the existing PR :)
Thanks for you motivation :v:

**Comment by JulianMaurin** [comment]
I take the `docs/contributing.md` 💪

**Comment by Jefidev** [comment]
Hi,
Ok I will also check the PR. I just want to be clear on some convention. Do you translate python types.
For instance Dict => dictionnaires
I did it but as a programmer even if my main language is french I refer to them as Dict.

For very specific technical terms I decided to do the same thing as the Spanish translator. They translate the terms but add an <abbr> note with the English technical term.

Here are the technical terms for which I cannot find a propper translation : 
- In-process background tasks.
- path operations

**Comment by ClemSau** [comment]
I am taking on `docs/benchmarks.md`.

**Comment by JulianMaurin** [comment]
I'll work on `docs/fastapi-people.md` :)

**Comment by JulianMaurin** [comment]
The next one is: `help-fastapi.md` 🚀

**Comment by JulianMaurin** [comment]
Hi @Jefidev 

> I just want to be clear on some convention. Do you translate python types.
For instance Dict => dictionnaires
I did it but as a programmer even if my main language is french I refer to them as Dict.
For very specific technical terms I decided to do the same thing as the Spanish translator. They translate the terms but add an note with the English technical term.

IMHO.
There is no reason to translate python keywords (eg. `dict`, `package`, `method`, ...).
In the other and, we have to translate the English word `dictionary` even when it's used in a technical context.

For the specifics technicals terms, you have to identify if there is an equivalent term in french. But most of the time when we are speaking about technical concepts, the usage of English could not be avoided/

My idea about a translated technical documentation is that it's just to make it easier for people to get into the library. We can assume that the reader will have a technical background and will not be confused by the presence of English technical terms.

**Comment by Jefidev** [comment]
Hi :smiley: 
I did not translate the words making reference to a python keyword (as you mentioned them `dict`, `package`, etc.) or the ones that are common in programming.

For the technical terms, I decided to take a look at the Spanish translation (because it is the most advanced yet but we are coming). They decided to translate the technical terms and display the English terms when hoovering (with the `<abbr>` tag). I follow that convention for my `features.md` translation.
For example, I decided to translate *Background Tasks* by *Tâches d'arrière plan*. But I decided to keep the words *path* in English as they should be familiar to the people designing API.

I am keen to discuss this issue more in-depth in order to have a strong convention for the French Translation Team.

**Comment by JulianMaurin** [comment]
And now `tutorial/index.md` 💪

**Comment by JulianMaurin** [comment]
> I am keen to discuss this issue more in-depth in order to have a strong convention for the French Translation Team. 

Yes, perfect. Let's confront our choices during the reviews, and report the conventions here :) 
(by the way, i'll do a small break in translations for now to do some reviews 👀 )

**Comment by rjNemo** [comment]
Hello,

what is the current status of the French translation?

I would like to contribute starting with `docs/alternatives.md`

**Comment by rjNemo** [comment]
Super. I will give them a look too

**Comment by Smlep** [comment]
Hi,

I'll work on `tutorial/background-tasks.md`.

**Comment by Smlep** [comment]
Hey 👋,

I'll work on `tutorial/external-links/md` now.

**Comment by NastasiaSaby** [comment]
Hello 👋 😃. 

I can do tutorial/testing.md.

**Comment by Smlep** [comment]
Hey 👋,

I'll translate `python-types.md`.

**Comment by Smlep** [comment]
Hi 👋,

I'll go for `project-generation.md` now!

**Comment by Smlep** [comment]
Hello 😁,

I'll translate `async.md`.

**Comment by rjNemo** [comment]
Hi 👋 ,
I translate `history-design-future.md`

**Comment by Smlep** [comment]
Hey 👋

I'll work on `tutorial/first-steps.md` now.

**Comment by Smlep** [comment]
I'll go with `tutorial/path-params.md` now 👋.

**Comment by Smlep** [comment]
Hey  👋

Now I'll work on `tutorial/query-params.md`

**Comment by Smlep** [comment]
Hi 👋 

Here is the translation for `tutorial/body.md`.

https://github.com/tiangolo/fastapi/pull/3671

**Comment by rjNemo** [comment]
Hi.

I translate Deployment starting with `deployment/index.md`

**Comment by Smlep** [comment]
Hi 👋

I'll go with `tutorial/query-params-str-validations.md` now.

**Comment by papysam** [comment]
Hello, I am sorry if I am in the wrong place I don't know where to report this kind of things.
There is a small error on this page : https://fastapi.tiangolo.com/fr/tutorial/body/
In the tip box about pycharm and the pydantic pycharm plugin at the chapter "Support de l'éditeur". The link don't redirect to the plugin (https://github.com/koxudaxi/pydantic-pycharm-plugin/) as it should but to pycharm (https://www.jetbrains.com/pycharm/)

**Comment by Smlep** [comment]
Hey @papysam 👋

Indeed, that was my bad. Nice catch 👍

This should fix it: https://github.com/tiangolo/fastapi/pull/4332

**Comment by JulianMaurin** [comment]
Bonjour, 

So good to see how the work progressed since my last visit here. Thanks a lot and great jobs to the contributors and reviewers 👏 
Sorry for such inactivity. I'm back on track and motivated to give some energy to this work. I think we can a

I think we can do a complete translation in a reasonable amount of time with two/three sufficiently motivated people (of course, the more we are, the faster it will be).

I will summarize the good practice on which we agree in the PR description to help newcomers. I've also some suggestions to make the review easier, I will detail that in another message.

Once again, well done, let's kick it out 🚀

**Comment by rjNemo** [comment]
It’s nice to see you back here!

**Comment by JulianMaurin** [comment]
Issue description updated, open to suggestions 🖖

**Comment by axel584** [comment]
Bonjour,
J'aimerai aider à la traduction de la documentation de FastAPI, mais je ne m'y connais pas trop en projet Open Source, j'ai plutôt l'habitude de contribuer sur un repo unique où j'ai des droits d'écriture. Du coup, je me posais pas mal de questions sur les review de PR : 
- Faut-il corriger/suggérer des modifications avec des commentaires dans la PR ? ou bien faire un commit de nos modifs ? (si oui, faut-il forker le repo du traducteur ou celui de tiangolo ?)
- Comment "approuver" une PR ? Faut-il des droits particuliers ? Je ne vois rien qui semblerait me permettre d'approuver des PR, ni sur celles ouvertes, ni sur celles déjà mergées.
- Y'a t'il des pages de doc à traduire en priorité ? (j'ai l'impression que certaines pages importantes semblent bloquées en attente d'approbation. Quand je recherche "french" dans les PR encore ouvertes, j'en trouve 13)

J'ai essayé de trouver des conseils un peu partout, mais en général, ça explique surtout le workflow fork/code/merge, et assez peu la partie "approbation d'une PR" par la communauté. Merci pour vos conseils.

**Comment by axel584** [comment]
I just translate `docs/fr/docs/advanced/additional-status-code.md`. But I've small problem with codecov... some test don't pass and I don't know why...

**Comment by rjNemo** [comment]
Hi @axel584, to have an idea of how to contribute the best would be to review open PRs.
You will see what files to update and what terms we usually use.

To approve a PR, go to the 'Files changed tab', then click on 'review changes' green button, and select 'approve' before submitting your review.

**Comment by axel584** [comment]
I wanted to translate `advanced/additional-responses.md` but when I commit/push my translation, this updated my first PR... So, I think it's better not to translate more in the meantime...

About review, I approve some of them (about deployment) but the others are pretty old... and seem to have conflict.

**Comment by JulianMaurin** [comment]
Hey @axel584 :wave: 
Thanks for you motivation! I'll have a look at your first translation :) 

>  when I commit/push my translation, this updated my first PR

Did you create a new branch? Are you comfortable with git?

> About review, I approve some of them (about deployment) but the others are pretty old... and seem to have conflict.

Do not hesitate to ping people on the PR to signal the conflict or to ask about news :)

**Comment by axel584** [comment]
@JulianMaurin : thank you for your advice. I'm not used to working on projects with pull requests between different repositories. But thanks to your advice, I should be able to do better in the future. The process explained seems very clear and very practical. I'm sorry for not being more attentive. The ability to compare line by line is really great to be able to check a translation. Thanks again !

**Comment by axel584** [comment]
I'll translate the `advanced/index.md` (and i'll try to follow the good practice :sweat_smile: )

**Comment by axel584** [comment]
@JulianMaurin / @rjNemo : I need a review for my PR and I didn't understand if I forgot something : https://github.com/tiangolo/fastapi/pull/5477
Thank your for your help or yours advices.

**Comment by axel584** [comment]
I will work on the files of the `advanced` section. Do not hesitate to contact me if you wish to translate pages from this section. I will take the pages in the order in which they appear on the documentation.

**Comment by frabc** [comment]
Bonjour @JulianMaurin / @rjNemo 👋

I would like to help you with the translation of the FastAPI documentation into French 😊

After reading the issue and the different best practices, I saw that there were still some PRs not merged (11 in total). Is there any need for review? 

I can also start translating pages, like for example with `tutorial/debugging.md`?

Thanks for your help 🙏

**Comment by rjNemo** [comment]
Hi @frabc 👋 ,
Sure, reviews would be very appreciated.

Then feel free to start translating 😄

**Comment by frabc** [comment]
Hi @rjNemo 👋

Thanks for the answer 🙂 A PR to be taken in priority or can I take the list of PR still open?

**Comment by frabc** [comment]
I'll work on `docs/tutorial/debugging.md` :)

**Comment by axel584** [comment]
Don't hesitate to ping me to review your PR.
You'll need 2 reviewer to approve your work.

Axel
